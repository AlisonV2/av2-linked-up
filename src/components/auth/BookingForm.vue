<template>
  <form class="auth-form" @submit.prevent="setBooking">
    <div class="row">
      <div class="col-12 text-center mb-4">
        <h2 class="title">Book a Stand</h2>
      </div>
    </div>
    <div class="form-floating mb-3">
      <input
        type="email"
        class="form-control"
        placeholder="Please confirm your email"
        v-model="email"
        required
      />
      <label>Please confirm your email</label>
    </div>
    <div class="error mb-3 mt-3">{{ error }}</div>
    <div class="btn-center">
      <app-button>Confirm</app-button>
    </div>
    <slot />
  </form>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports BookingForm
 * @type {Component}
 * @requires Sentry
 * @vue-data {number} emails - v-model
 * @vue-event resetPassword
 */
export default {
  props: ['id'],
  data() {
    return {
      email: '',
      error: null,
    };
  },
  methods: {
    setBooking() {
      const data = {
        eventId: this.id,
        email: this.email,
      }
      this.$store
        .dispatch('setBooking', data)
        .then(() => {
          this.$router.push({ name: 'AdminEvents' });
        })
        .catch((err) => {
          this.error = err.message;
          Sentry.captureException(err);
          return;
        });
    },
  },
};
</script>

<style lang="scss" scoped>
.title {
  color: $dark;
  font-size: 1.8rem;
  margin-bottom: 1.5rem;
}
</style>
