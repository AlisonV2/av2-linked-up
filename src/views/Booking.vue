<template>
  <div class="container col-6 home">
    <div>
      <BookingForm :id="id" />
    </div>
  </div>
</template>

<script>
import BookingForm from '@/components/auth/BookingForm';

/**
 * @exports BookingPassword
 * @requires BookingForm
 * @type {Page}
 */
export default {
  name: 'BookingPassword',
  components: {
    BookingForm,
  },
  data() {
    return {
      id: this.$route.params.id,
    };
  },
};
</script>
