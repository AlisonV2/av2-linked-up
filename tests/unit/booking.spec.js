import { shallowMount } from '@vue/test-utils';
import Booking from '@/views/Booking.vue';
import BookingForm from '@/components/auth/BookingForm';

const $route = {
  params: { id: '1' },
};
const config = {
  data() {
    return {
      id: '1',
    };
  },
  global: {
    components: {
      BookingForm,
    },
    mocks: $route,
  },
};

/**
 * @module BookingPageTest
 */
describe('Booking.vue', () => {
  it('BookingPage Snapshot', () => {
    const component = shallowMount(Booking, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if BookingForm is rendered', () => {
    const component = shallowMount(Booking, config);
    const form = component.findComponent(BookingForm);
    expect(form.exists()).toBe(true);
  });
});
