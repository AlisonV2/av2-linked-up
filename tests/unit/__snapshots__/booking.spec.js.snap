// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking.vue BookingPage Snapshot 1`] = `
<div
  class="container col-6 home"
>
  <div>
    <booking-form-stub
      id="1"
    />
  </div>
</div>
`;
